python/foglamp/services/core/api/service.py

```diff
@@ -95,72 +96,105 @@ def get_service_records():
-        schedule.name = name
-        schedule.process_name = name
-        schedule.repeat = datetime.timedelta(0)
-        schedule.exclusive = True
-        schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
-
-        # Save schedule
-        await server.Server.scheduler.save_schedule(schedule, is_enabled)
-        schedule = await server.Server.scheduler.get_schedule_by_name(name)
+        # If all successful then lastly add a schedule to run the new service at startup
+        try:
+            schedule = StartUpSchedule()
+            schedule.name = name
+            schedule.process_name = name
+            schedule.repeat = datetime.timedelta(0)
+            schedule.exclusive = True
+            schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
+
+            # Save schedule
+            await server.Server.scheduler.save_schedule(schedule, is_enabled)
+            schedule = await server.Server.scheduler.get_schedule_by_name(name)
+        except StorageServerError as ex:
+            await revert_configuration(name)  # Revert configuration entry
+            await revert_scheduled_processes(name)  # Revert scheduled_process entry
+            raise web.HTTPInternalServerError(reason='Failed to created schedule. {}'.format(ex.error))
+        except Exception as ins_ex:
+            await revert_configuration(name)  # Revert configuration entry
+            await revert_scheduled_processes(name)  # Revert scheduled_process entry
+            raise web.HTTPInternalServerError(reason='Failed to created schedule. {}'.format(str(ins_ex)))
 
         return web.json_response({'name': name, 'id': str(schedule.schedule_id)})
 
     except ValueError as ex:
         raise web.HTTPNotFound(reason=str(ex))
 
+
+async def check_schedules(schedule_name):
+    storage = connect.get_storage_async()
+    payload = PayloadBuilder().SELECT("name").WHERE(['name', '=', schedule_name]).payload()
+    result = await storage.query_tbl_with_payload('scheduled_processes', payload)
+    return result['count']
+
+async def check_scheduled_processes(process_name):
+    storage = connect.get_storage_async()
+    payload = PayloadBuilder().SELECT("schedule_name").WHERE(['schedule_name', '=', process_name]).payload()
+    result = await storage.query_tbl_with_payload('schedules', payload)
+    return result['count']
+
+async def revert_scheduled_processes(process_name):
+    storage = connect.get_storage_async()
+    payload = PayloadBuilder().WHERE(['name', '=', process_name]).payload()
+    result = await storage.delete_from_tbl('scheduled_processes', payload)
+
+async def revert_configuration(key):
+    storage = connect.get_storage_async()
+    payload = PayloadBuilder().WHERE(['key', '=', key]).payload()
+    result = await storage.delete_from_tbl('configuration', payload)
```

Comment:
its result does not matter? as it has no return. Also please a line-break at the end of the file